### DISCUSSION TITLE
Is Focus still needed/useful?

### DISCUSSION THREAD

**Discussion by Salty-Panda** [CleverRaven/Cataclysm-DDA#48758] [General]
Inspired by @actual-nh 
> As an educator (adjunct professor) for well over 20 years, I have to say that the mechanism of "disable learning some skills so your focus doesn't get removed" is... one of the least realistic in the game, to be blunt. Much of learning - at least, long-term successful learning - is making connections with what you already know. Learning more things doesn't hurt your learning other things. The closest to realistic this mechanism gets is when someone is completely overwhelmed with new things... but one doesn't overcome that by shutting off learning of some/most of it; one overcomes it by seeing the connections between new things - and between new things and old things you already know - and realizing that it's not all that much completely new stuff in the first place. Another problem with the current setup is that it fails to take into account the morale boosts that come with successfully learning new things (including by seeing how things you already know interrelate).

AFAIK, Focus's purpose is to limit how quickly player can grow in power
With proficiencies, tiredness and weariness acting as natural limiters to player's activities, is it still needed?

**What can do player with:**
(by "having" something I mean 100 Focus, Fresh weariness and lack of tiredness/sleepiness)
- tiredness and weariness:
anything, except he's unable to gain any experience from that, which is rather unrealistic
Actually, in terms of training combat skills Focus is zeroed nearly instantly but skill gain for them isn't as heavily impaired as in case of other ones. At least it's my observation, since you still can train dodge by standing still surrounded by enemies.
- focus and tiredness:
Can perform light activities like reading, making food and level affiliated skills quickly
- focus and weariness
Tiredness indirectly lowers Focus via mood penalty
Can do anything, but with some penalty to crafting speed and (not sure about this one) reading speed.
- focus
Tiredness indirectly lowers Focus via mood penalty
Can do light activities with penalties.

Worth of noting, Focus loss for reading is twofold.
Primary caused by natural loss following skill gain and secondary caused by negative enjoyment value of most skill training books which stacks very fast.

So what's the purpose of Focus here if it's either artificial of duplicates effect of other modifiers?
It's indicator for player's ability to train skills visible for user. A single coefficient calculated from most of other modifiers.
But why different treatment for combat skills, why it oscillates so quickly?

Suggestion:

- Focus could be removed with possible global nerf/balancing to learning speed and it's current role of modifier to skill gain could be fulfilled by currently existing natural limiters of player's activity and long-term morale.
- Focus could be changed into a read-only value, not being influenced directly and only serving to display compounded effects of all other factors

I know there was some Discussion/PR/FR featuring long-term morale system involving components like security, traumas etc but can't find it.

**Comment by actual-nh** [comment]
I know that the current focus system is not quite as @kevingranade wants it to be long-term. I can see the concept of something like Focus - but as a measure of _**mental** tiredness_, paralleling stamina and weariness, without such a drain to it from learning in specific.

**Reply by I-am-Erk** [reply]
Exactly this. I think we'll be revisiting focus soon. It definitely represents something we want, but it's clunky at the moment.

**Comment by Daimiou** [comment]
I might be one of the few who actually like the focus system. It's something you can wholly ignore and keep going with little affect to your enjoyment of the game and overall progress. But if you want to minmax aim for efficiency then it's something to keep an eye on and keep a good care of, after a while it (keeping track of focus) becomes almost automatic and stops being a chore.

The way I look at focus, is a numeric measurement of your ability to pay attention to detail or ability to mentally multitask. The same way as you try to solve a complex problem (math for example), if not proficient or not used to, it'll quickly tire you and make it hard to focus on it, you'll then need to take a short break and relax to gather thoughts and rethink the approach. Ever got stuck on a hard boss in some game, just to clear it easily after taking a break from it?

So in game as you get more skilled and become more proficient at doing something it drains less and less focus to the point of not using it at all. The problem with focus as it is in my opinion is just how badly it is explained (not at all really). It is just another numeric representation of something, and worst, something so nebulous as the said ability to focus/pay attention/multitask. 

Honestly, adding it and implementing it even the way it is now, was a stroke of genius on part of @kevingranade.

Gameplay issue with it as of now is, as I see it, two-fold.

First: the inconsistent time flow in game against real world time flow: most of the time we're operating on second to second time frames, then suddenly we're having significant time skips during complex automated tasks which throws off people. This is something I do not think can be unfortunately fixed/mitigated in any way, except for explaining/getting used to it.

Second: it's interaction with other activity/exertion measuring mechanics: frankly, the only thing the focus use should affect is weariness (mental exertion) and only in a small quantity, since focus relates to the immediate and can be gone as fast as it can be regained, with just a short break, where weariness which relates mostly to more intellectual/unfamiliar  (lack of proficiency) tasks, it's drained steadily as you remain active with rates depending on type and level of activity. As weariness grows it should then affect either rate the focus is regained at or simply progressively cap maximum focus. Tiredness (what I consider physical exertion) should not have any affect on focus, rather keep its interaction only to weariness. As to morale (or rather mood): I'd reduce it's affect on focus or let it be modified with some traits, being ungodly pissed sometimes lets you focus just as well if not better on the task at hand. All in all, it's just a matter of number tweaking and testing the flow.

Those are my two cents on the matter, I'm sure I missed something crucial in the rant but I'm out of focus and gonna take a break.

**Reply by actual-nh** [reply]
Currently, weariness is essentially tracking caloric expenditure (it may be considered a measurement of liver+muscle glycogen, for instance); it is not related to lack of proficiency (except insofar as failing may make you take longer at a task). Weariness is, I think, thus closer to what you're talking about as "tiredness".

The combination of Focus and Morale is the closest thing currently to measuring mental exertion, and IMHO it's in a rather unclear manner. I agree that Morale's current effects on Focus could stand to be refined - although this may require splitting Morale into something like a mad/sad/glad emotional distinction.
